community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/legacystore/v21/propertydeduplication/PropertyDeduplicator.java

```diff
@@ -176,13 +179,16 @@ private void resolveConflicts(
         // First find and resolve the duplicateClusters for all properties whose nodes are indexed.
         // The duplicateClusters are indexed by the propertyRecordId of the head-record in the property chain, so any node
         // whose nextProp() is amongst our duplicateClusters is potentially interesting.
-        try ( IndexLookup indexLookup = new IndexLookup( schemaStore, schemaIndexProvider );
-              IndexedConflictsResolver indexedConflictsResolver =
-                      new IndexedConflictsResolver( duplicateClusters, indexLookup, nodeStore, propertyStore ) )
+        if (isIndexStorageNotEmpty( storeDir ))
```

Comment:
Please don't have negations in method names and meanings like this. Much better with: `if ( indexStorageEmpty( storeDir ) )` or similar.